### PR TITLE
helm: make quota accounting configurable

### LINKED ISSUE #562: helm: configurable quota accounting behaviour
**Current behaviour**

When somebody installs 0.8.0-alpha.4, by default the quota system is not imposing any limits on users:

```console
$ reana-client quota-show --resource cpu --report limit   
No limit.
$ reana-client quota-show --resource disk --report limit
No limit.
```

However, when a user runs a few workflows, the quota consumption is being calculated:

```console
$ reana-client quota-show --resource cpu --report usage
60328
$ reana-client quota-show --resource disk --report usage   
1577932
```

This causes troubles with workflow runtime pod termination, see https://github.com/reanahub/reana-db/issues/149.

As an administrator, if you would like to deploy a system that does not care about quotas, you currently cannot without wild patches such as https://github.com/reanahub/reana-db/pull/150#discussion_r733896652

**Expected behaviour**

It should be possible to deploy a REANA system easily for the use case of no limits, no quota calculations. This means to introduce a new easy-to-modify helm value that could alter quota usage calculation policy, e.g. do it after every workflow run (i.e. current behaviour), e.g. do it nightly (i.e. via crontab-like updater, as we do now; but not as part of termination tasks, which is not scalable); e.g. don't do it at all (i.e. completely turned off, for the fastest termination tasks, as in the wild patch above). Note that various policies might also affect how quota usage is being displayed (or not) in the web UI, for the sake of consistency.

See also https://github.com/reanahub/reana-db/issues/149

### PR DESCRIPTION
**Pull request by mvidalgarcia**
closes #562

### CODE REVIEW

**Review by audrium — COMMENTED** [review]
Disabling quota calculation works well!

Few minor remarks:
- Right now it's possible to either disable quota calculations at all, or to do it after every workflow run + cronjob. We are missing a setting to do it only via cronjob.
- `quota.termination_update_policy` configuration name is not really clear, since it's not a policy, but more like a resource list for quota calculation (which is used also in cronjob, so not only `termination`?)

**Comment by mvidalgarcia** [comment]
Thanks for the review! I enriched the logic a little bit to be able to have more configurations.

>   * Right now it's possible to either disable quota calculations at all, or to do it after every workflow run + cronjob. We are missing a setting to do it only via cronjob.

I made it more fine-grained to support enabling only the cronjob or only the workflow terminantion logic.
One now can:
- Disable all the quota functionality altogether by switching the `enabled` flag to `false`.
- When quota is `enabled: true`:
  - Disable only the workflow termination accounting by not setting `termination_update_policy`.
  - Disable only the cronjobs by not setting the `disk_update` crontab.
 
>   * `quota.termination_update_policy` configuration name is not really clear, since it's not a policy, but more like a resource list for quota calculation (which is used also in cronjob, so not only `termination`?)

The name is based on the [`REANA_WORKFLOW_TERMINATION_QUOTA_UPDATE_POLICY` env var](https://github.com/reanahub/reana-db/blob/9734af69f57e5ec425150c690552f8c179681343/reana_db/config.py#L91), and now it only refers to the actual workflow termination updater (no cronjob).
Do you have another name in mind? I guess that if we change it we should also change the env var name in reana-db?

**Review by audrium — APPROVED** [review]
Tested carefully, everything works nicely!

- Left one minor comment for the readme file
- > The name is based on the REANA_WORKFLOW_TERMINATION_QUOTA_UPDATE_POLICY env var, and now it only refers to the actual workflow termination updater (no cronjob).

   Make sense and looks good to me since it's now only related with workflow termination

File: helm/reana/README.md [hunk 1]
```diff
@@ -98,7 +98,9 @@ This Helm automatically prefixes all names using the release name to avoid colli
 | `traefik.enabled`                                        | Install Traefik in the cluster when installing REANA                                 | true                                            |
 | `volume_paths.root_path`                                 | Path to the REANA directory inside the underlying storage volume                     | /var/reana                                      |
 | `volume_paths.shared_volume_path`                        | Path inside the REANA components where the shared volume will be mounted             | /var/reana                                      |
+| `quota.enabled`                                          | Enable user workflow accounting capabilities.                                        | true                                            |
 | `quota.disk_update`                                      | Cronjob pattern representing how often the users disk quota usage should be updated. Leave it empty to deactivate it | "0 3 * * *"     |
+| `quota.termination_update_policy`                        | Resources to calculate quotas on worflow termination. Possible values: "cpu" and "disk" | "cpu,disk"                                   |
```
[the remark this answers is not in the document]
Comment (mvidalgarcia) [reply helm/reana/README.md:103]: Yes, it makes total sense, adding it 👍

**Pull request merged**